### ISSUE TITLE
Why your model generated mnist images are noises?

### ISSUE THREAD

**Issue by CatLoves** [explainingai-code/StableDiffusion-PyTorch#4]
Dear explainingai-code:
    Your codebase for ddpm is so detailed and helpful that I would like to thank you very much for your great work ! 
    I have downloaded this codebase and follow your instructions carefully, luckly I got a good VQVAE result as follows:
![image](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/7496455/5b807cd3-c1af-4d7f-af6d-77e37275bac7)

    So I continue to train unconditional and class conditional ddpm models. 
    I run python tools/train_ddpm_vqvae.py to train unconditional ddpm model and it seems to have converged as follows:
![image](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/7496455/8a15df65-f514-49ef-bae2-424fe058aaca)
     The loss decreases from 0.2833 to 0.0886. Then I run python tools/sample_ddpm_vqvae.py to check model output, I get mnist/samples/x0_0.png to x0_999.png, all seems like PURE NOISE as follows:
![image](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/7496455/b2583117-969f-4385-9714-46c789249b32)

     Simiarly I trained and tested mnist class conditional ddpm and all results seems like PURE NOISE as follows:
![image](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/7496455/f7418e3f-f950-42d1-a2d5-189488f6f2ff)

     My code structure as follows:
![image](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/7496455/db341701-10f1-45f4-a9af-77789d0bc4af)
     I am sure that I just follow your instructions and DO NOT modify model related code, but the results seems weired. Appreciated if you could give some help.

Sincere,
CatLoves

**Comment by explainingai-code** [comment]
Hello @CatLoves, Thank you for the appreciation. 
Can you please share both the x0_0.png and x0_999.png file. The x0_0.png will be a decoded image and x0_999.png will be a pure noise(for unconditional) but for conditional x0_999.png will still be something that looks like digits(like the ones you uploaded).

**Comment by CatLoves** [comment]
@explainingai-code Sure. The x0_0.png under mnist/samples is as follows:
![image](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/7496455/73ebbbe8-351f-4734-b81e-f641f58911d6)
 It seems very nice. So for this problem the reason is that I didn't figure out that only the last step's output would be decoded, x0_1.png to x0_999.png would be decoded latent rather than image itself. Sorry for this misunderstanding. 
 For class conditional ddpm on mnist, the x0_0 is as follows:
![image](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/7496455/0b5162be-2263-49dd-9d79-4d47cdfede3d)
 It also makes sense. 
 Thank you again for your quick and detailed response. I am looking forward to your further great work. Have a nice day !

**Comment by explainingai-code** [comment]
Thats great! I understand how the image numbering could be confusing. Have adding this in the readme output section now.

Thank You
